beastling/util/misc.py

```diff
@@ -0,0 +1,56 @@
+import newick
+
+
+def sanitise_tree(tree, tree_type, languages):
```

Comment:
I think stuff like this should really live as helper function in some utility module. As method in such a central class as `Configuration` it takes up just too much screen to make the classes functionality transparent.

Reply:
Definitely!  I have long been unhappy with how huge `Configuration` is.

But one problem with factoring stuff out into helper functions is that it becomes cumbersom to pass messages to the user.  This already troubles me with our CLDF loading stuff - it's in a function by itself, and can't easily append to the calling model's `messages`.  Probably the tree sanitizer is unlikely to want to leave log messages, but it's a pain point in general.

But then, perhaps collecting messages in a list to print at the end is a silly approach (it's not nice when things crash before the print).  Maybe we need a utility `log` function which reads some global flag, set by the CLI, to determine whether to print the stuff it is passed or stay quiet for when BEASTling is used as a library...

Reply:
I experimented a bit more with logging, and I think we should start with a simple 
```python
import logging
log = logging.getLogger(__name__)
```
in each module (see https://docs.python-guide.org/writing/logging/#logging-in-a-library) and then think about how we want to support the user with configuration, i.e. think about https://docs.python-guide.org/writing/logging/#logging-in-an-application

Reply:
Note, though, that the `logging.getLogger` call shouldn't be on module level, though, but happen within functions to avoid getting un-configured loggers.

Reply:
Proper logging configuration will also give us the `urgent_messages` vs `messages` distinction for free. - Although I just realized that `urgent_messages` isn't appended to anywhere ...

Reply:
Whoops!  I guess at one point I thought something should be upgraded to "urgent" status but never actually went ahead and did it after setting up the infrastructure.  But proper logging does indeed seem like the way to go.

Reply:
This (a switch to proper logging) would be also be slightly relevant to #91.